api_tests/users/views/test_user_detail.py

```diff
@@ -995,3 +994,228 @@ def test_requesting_deactivated_user_returns_410_response_and_meta_info(
+        assert res.json['errors'][0]['detail'] == "For 'startYear' the field value -2 is less than the minimum of 1900"
+
+    def test_user_put_profile_date_validate_ongoing_position(self, app, user_one, user_one_url, request_payload, request_key):
+        # endDates for ongoing position
+        request_payload['data']['attributes'][request_key][0]['ongoing'] = True
+        del request_payload['data']['attributes'][request_key][0]['endYear']
+        res = app.put_json_api(user_one_url, request_payload, auth=user_one.auth, expect_errors=True)
+        assert res.status_code == 400
+        assert res.json['errors'][0]['detail'] == "For 'ongoing' the field value True is not valid under any of the given schemas"
+
+    def test_user_put_profile_date_validate_end_date(self, app, user_one, user_one_url, request_payload, request_key):
+        # End date is greater then start date
+        request_payload['data']['attributes'][request_key][0]['startYear'] = 2000
+        res = app.put_json_api(user_one_url, request_payload, auth=user_one.auth, expect_errors=True)
+        assert res.status_code == 400
+        assert res.json['errors'][0]['detail'] == 'End date must be greater than or equal to the start date.'
+
+    def test_user_put_profile_date_validate_end_month_dependency(self, app, user_one, user_one_url, end_month_dependency_payload):
+        # No endMonth with endYear
+        res = app.put_json_api(user_one_url, end_month_dependency_payload, auth=user_one.auth, expect_errors=True)
+        assert res.status_code == 400
+        assert res.json['errors'][0]['detail'] == "u'endYear' is a dependency of u'endMonth'"
+
+    def test_user_put_profile_date_validate_start_month_dependency(self, app, user_one, user_one_url, start_month_dependency_payload):
+        # No endMonth with endYear
+        res = app.put_json_api(user_one_url, start_month_dependency_payload, auth=user_one.auth, expect_errors=True)
+        assert res.status_code == 400
+        assert res.json['errors'][0]['detail'] == "u'startYear' is a dependency of u'startMonth'"
+
+    def test_user_put_profile_date_validate_start_date_no_end_date_not_ongoing(self, app, user_one, user_attr, user_one_url, start_dates_no_end_dates_payload, request_key):
+        # End date is greater then start date
+        res = app.put_json_api(user_one_url, start_dates_no_end_dates_payload, auth=user_one.auth, expect_errors=True)
+        user_one.reload()
+        assert res.status_code == 400
+        assert res.json['errors'][0]['detail'] == "For 'ongoing' the field value True is not valid under any of the given schemas"
+
+    def test_user_put_profile_date_validate_end_date_no_start_date(self, app, user_one, user_attr, user_one_url, end_dates_no_start_dates_payload, request_key):
+        # End dates, but no start dates
+        res = app.put_json_api(user_one_url, end_dates_no_start_dates_payload, auth=user_one.auth)
+        user_one.reload()
+        assert res.status_code == 200
```

Comment:
The frontend doesn't currently allow this:
![screen shot 2018-07-30 at 2 37 44 pm](https://user-images.githubusercontent.com/7913604/43416414-2ba7b106-9406-11e8-9599-476fd0b9e35b.png)